pkg/alertmanager/alertmanager.go

```diff
@@ -241,7 +242,38 @@ func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
 		am.wg.Done()
 	}()
 
-	am.alerts, err = mem.NewAlerts(context.Background(), am.marker, 30*time.Minute, nil, am.logger)
+	var callback mem.AlertStoreCallback
+	if am.cfg.Limits != nil {
+		insertAlertFailures := promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "alertmanager_insert_alert_failures_total",
```

Comment:
```suggestion
			Name: "alertmanager_alerts_insert_failures_total",
```

Reply:
To keep the `alertmanager_alerts` prefix homogenous?

Reply:
sounds good